MsCorePkg/Library/MathLib/MathLib.c

```diff
@@ -218,7 +218,7 @@ sqrt_d (
   IN CONST double  input
   )
 {
-  UINT64  firstGuess = (UINT64)input;
+  UINT32  firstGuess = (UINT32)input;

```

Comment:
I might be missing something on this, but wouldn't this cast cause the upper 32 bit getting truncated as a double should be 64bit in width?

Reply:
This is exactly why I asked for a review, while this fixed building, it's not the best solution, hence I need someone with experience to look at this

Reply:
@sonic011gamer, looking at your build log `__aeabi_ul2d` looks like a compiler built-in function that is not defined. We define those for Arm an AARCH64 in https://github.com/microsoft/mu_basecore/tree/release/202302/MdePkg/Library/CompilerIntrinsicsLib.

That usually gets linked to every module by applying it like this in a DSC:
`NULL|MdePkg/Library/CompilerIntrinsicsLib/ArmCompilerIntrinsicsLib.inf`

Could you try defining that function there?

Reply:
It looks like there is a definition in `ArmSoftFloatLib` in https://github.com/microsoft/mu_silicon_arm_tiano/tree/release/202302/ArmPkg/Library/ArmSoftFloatLib. Are you linking that against this module?

Reply:
> It looks like there is a definition in `ArmSoftFloatLib` in https://github.com/microsoft/mu_silicon_arm_tiano/tree/release/202302/ArmPkg/Library/ArmSoftFloatLib. Are you linking that against this module?

yes, I am building with this module and the one above mentioned included in the .dsc